pulsar/consumer.go

```diff
@@ -305,8 +325,26 @@ type Consumer interface {
 	Ack(Message) error
 
 	// AckID the consumption of a single message, identified by its MessageID
+	// When `EnableBatchIndexAcknowledgment` is false, if a message ID represents a message in the batch,
+	// it will not be actually acknowledged by broker until all messages in that batch are acknowledged via
+	// `AckID` or `AckIDList`.
 	AckID(MessageID) error
 
+	// AckIDList the consumption of a list of messages, identified by their MessageIDs
+	//
+	// This method should be used when `AckWithResponse` is true. Otherwise, it will be equivalent with calling
+	// `AckID` on each message ID in the list.
+	//
+	// If there are some invalid message IDs in the list, this method won't return an error for them, only error logs
+	// will be printed because retrying acknowledging them will not make a difference. Message IDs retrieved from
+	// Message objects are always valid.
+	//
+	// If the consumer has subscribed multiple topics or partitions, since each topic is associated with an ACK request,
+	// `AckError` will be returned if there are errors because some message IDs might succeed while others might fail.
+	// If you're sure that only 1 message is subscribed, you should treat all message IDs as failed when it returns an
```

Comment:
> If you're sure that only 1 message is subscribed, you should treat all message IDs as failed when it returns an error

Why is this logic needed? We could keep it simple and return AckError even if only one topic has an error.